### ISSUE TITLE
NmeaLineParser should include argument name in `ArgumentException`

### ISSUE THREAD

**Issue by idg10** [ais-dotnet/Ais.Net#152]
Also, the docs should state that the `NmeaLineParser` ctos throws an `ArgumentException` to report the discovery of bad data.

**Comment by fredeil** [comment]
Hey @idg10 ,  we're using this library (at @dualog) and wanted to contribute back.

Could you clarify if you would like to have all of the `ArgumentException` s to reference the `line` input parameter? It's the only argument for the method but some of the argumentExceptions are being thrown on a subset of that line

See https://github.com/fredeil/Ais.Net/commit/2655552f56055596fb2a1abf37ba71a7b2fe1463

**Comment by idg10** [comment]
Yes, I just meant that the relevant throws should use `nameof(line)` as the `paramName` argument when constructing the `ArgumentException`. (So exactly what you've done in that commit.)

Thanks for offering to do this. While you're in there would you mind adding an [`<exception>`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags?msclkid=8002abd4b57d11ec8acb9c3796f6a72a#exception) tag to the relevant XML doc comments, so it's clear to anyone using this library that this is how we report unparseable data?

**Comment by fredeil** [comment]
Sure, that was my next question. 

No more documentation needed outside of that? @idg10

**Comment by idg10** [comment]
Nothing I'm aware of, but if you feel it's missing something feel free either to add more, or to open another issue describing what's missing.
